### PR TITLE
#1055: Cobertura-maven-plugin doesn't generate xml report

### PR DESCRIPTION
**Pull request by 2686747**
#1055: : changed coveralls trigger, added cobertura profile, updated cobertura, coveralls versions

### CODE REVIEW

**Comment by karato** [comment]
@2686747 I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by karato** [comment]
@mkordas review it

**Comment by mkordas** [comment]
@2686747 I'm on it

**Comment by mkordas** [comment]
@2686747 can we also revert e228615082183ca036df79799b57e147f42dedc6 here?

File: pom.xml [hunk 1]
```diff
@@ -161,17 +161,22 @@
...
+                        <artifactId>cobertura-maven-plugin</artifactId>
+                        <version>2.7</version>
+                </plugin>
             </plugins>
         </pluginManagement>
     </build>
     <profiles>
-        <profile>
-            <id>coveralls</id>
+    <profile>
```
Comment (mkordas) [inline pom.xml]: @2686747 do we need to modify this line?

File: pom.xml [hunk 2]
```diff
@@ -194,6 +199,16 @@
                             </execution>
                         </executions>
                     </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>coveralls</id>
```
Comment (mkordas) [inline pom.xml]: @2686747 why do we need separate profiles for coveralls and cobertura?

**Comment by mkordas** [comment]
@2686747 can you explain why deployment was working before?

File: pom.xml [hunk 3]
```diff
@@ -194,6 +199,16 @@
...
+            <activation>
+                <file><exists>target/site/cobertura/coverage.xml</exists></file>
```
Comment (mkordas) [inline pom.xml]: @2686747 shouldn't report be always generated?

File: pom.xml [hunk 4]
```diff
@@ -182,7 +187,7 @@
                         <artifactId>cobertura-maven-plugin</artifactId>
                         <executions>
                             <execution>
-                                <phase>site</phase>
+                                <phase>pre-site</phase>
```
Comment (mkordas) [inline pom.xml]: @2686747 why do we need to change that?

**Comment by mkordas** [comment]
@2686747 please see my comments above

File: pom.xml (hunk 2 above)
Comment (2686747) [inline pom.xml]: Coveralls failed if cobertura didn't generate xml report. Splitting provides a possibilty for trigger (coveralls works now only if report is existed)

File: pom.xml (hunk 3 above)
Comment (2686747) [inline pom.xml]: Now cobertura always works, coveralls - only if there is target/site/cobertura/coverage.xml.

File: pom.xml (hunk 4 above)
Comment (2686747) [inline pom.xml]: Cobertura generates report before coveralls.

File: pom.xml (hunk 1 above)
Comment (2686747) [inline pom.xml]: @mkordas for splitting profiles

**Comment by 2686747** [comment]
@mkordas 

> @2686747 can you explain why deployment was working before?

I think coveralls profile was turned off.

**Comment by 2686747** [comment]
@mkordas  Thanks for review, see my comments, please.

File: pom.xml (hunk 2 above)
Comment (mkordas) [inline pom.xml]: @2686747 thanks, now I see

File: pom.xml (hunk 1 above)
Comment (mkordas) [inline pom.xml]: @2686747 but `<profile> existed before, you just changed indentation

File: pom.xml (hunk 3 above)
Comment (mkordas) [inline pom.xml]: @2686747 makes sense

File: pom.xml (hunk 4 above)
Comment (mkordas) [inline pom.xml]: @2686747 OK, good

**Comment by mkordas** [comment]
@2686747 just one comment about wrong indentation is left

**Comment by 2686747** [comment]
@mkordas i think about cobertura [version](https://github.com/jcabi/jcabi-parent/blob/master/pom.xml#L401). Localy i had errors with 2.6, but coveralls didn't fail. Upgrading fixed this, but  i think we can try to leave 2.6 version.

**Comment by 2686747** [comment]
@mkordas i think about cobertura [version](https://github.com/jcabi/jcabi-parent/blob/master/pom.xml#L401). Localy i had cobertura errors with 2.6, but coveralls didn't fail. Upgrading fixed this, but  i think we can try to leave 2.6 version.

**Comment by mkordas** [comment]
@2686747 OK, let's leave it then

**Comment by mkordas** [comment]
@2686747 looks good!

**Comment by mkordas** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova many thanks

**Comment by karato** [comment]
@mkordas since quality is good, 10 mins paid to @dmzaytsev (our architect), transaction `79352567`

**Comment by karato** [comment]
@mkordas paied 10 mins to @elenavolokhova for QA review (payment ID is `79354793`)... thank you, added **31 mins** to your acc, payment num is `56db3dcb825639136e000360`, 42 hours and 40 mins spent in total to complete... you're getting extra minutes here (c=16)... +31 added to your rating, at the moment it is: [+8157](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**